src/app/external-cluster-wizard/steps/external-cluster/provider/eks/component.ts

```diff
@@ -120,8 +126,12 @@ export class EKSClusterSettingsComponent
   securityGroupLabel = SecurityGroupState.Ready;
   securityGroups: EKSSecurityGroup[] = [];
   roleArn: EKSClusterRoleList[] = [];
-  selectedRoleArn = '';
   roleArnLabel = RoleArnState.Ready;
+  selectedRoleArn = '';
+  nodeRoles: EKSNodeRole[] = [];
+  selectedNodeRolArn = '';
+  nodeRoleLabel = NodeRoleState.Ready;
```

Comment:
Why do we need to create separate properties and types for this? These are exactly the same as `roleArn`. Only difference is the endpoints where these lists are fetched from.

Reply:
2 points:
- Cluster Service Role != Node I am Role
- Yes there might be duplication interms of Interface(Class) but both are sep things from backend point of view

Secondly, if you mean to say that we can do that based on if wizard mode call this func or set this label we can do that but haven't done adding a sep state will keep distinct from previous one Otherwise we need to rename previous handling of cluster service role (label, selected, class) to some generic name that can be used in both Cluster/MD

Reply:
We can keep these types but then we should rename `roleArn`, `selectedRoleArn` etc. to `clusterRoleArn` so its easier to differentiate.

Reply:
sounds good